lib/_tls_wrap.js

```diff
@@ -300,12 +300,16 @@ var proxiedMethods = [
 ];
 
 // Proxy HandleWrap, PipeWrap and TCPWrap methods
-proxiedMethods.forEach(function(name) {
-  tls_wrap.TLSWrap.prototype[name] = function methodProxy(...args) {
+function makeMethodProxy(name) {
+  return function methodProxy(...args) {
```

Comment:
I would vote for changing to `arguments` and `.apply()` usage while we're in here.